bouncer/shared/fill_or_kill.ts

```diff
@@ -0,0 +1,139 @@
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  amountToFineAmount,
+  amountToFineAmountBigInt,
+  assetDecimals,
+  decodeDotAddressForContract,
+  isWithinOnePercent,
+  newAddress,
+  observeBalanceIncrease,
+  shortChainFromAsset,
+} from './utils';
+import { requestNewSwap } from './perform_swap';
+import { send } from './send';
+import { getBalance } from './get_balance';
+import { getChainflipApi, observeEvent } from './utils/substrate';
+import { RefundParameters } from './new_swap';
+
+/// Do a swap with unrealistic minimum price so it gets retried and then refunded.
+async function testMinPriceRefund(asset: Asset, amount: number) {
+  const swapAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
```

Comment:
swapAsset is ambiguous
```suggestion
  const outputAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
```